source/extensions/tracers/xray/tracer.cc

```diff
@@ -122,16 +122,18 @@ Tracing::SpanPtr Span::spawnChild(const Tracing::Config&, const std::string& ope
   return child_span;
 }
 
-Tracing::SpanPtr Tracer::startSpan(const std::string& span_name, const std::string& operation_name,
-                                   Envoy::SystemTime start_time,
+Tracing::SpanPtr Tracer::startSpan(const std::string& operation_name, Envoy::SystemTime start_time,
                                    const absl::optional<XRayHeader>& xray_header) {
 
   const auto ticks = time_source_.monotonicTime().time_since_epoch().count();
   auto span_ptr = std::make_unique<XRay::Span>(time_source_, *daemon_broker_);
   span_ptr->setId(ticks);
-  span_ptr->setName(span_name);
+  span_ptr->setName(segment_name_);
   span_ptr->setOperation(operation_name);
+  // even though we have a TimeSource member in the tracer, we
+  // assume the start_time argument has more precise value than
```

Comment:
nit: please start with capital, end with period, and finish sentence.

Reply:
Good catch. Thank you.